olive/passes/onnx/optimum_conversion.py

```diff
@@ -70,42 +76,64 @@ def _run_for_config(
-        # so that we only keep the final model files in the output_model_path
-        # and track external data if present
-        config["extra_args"] = config["extra_args"] or {}
-        config["extra_args"].update(
+        extra_args = deepcopy(config["extra_args"]) or {}
+        extra_args.update(
             {
                 "opset": config["target_opset"],
                 "fp16": config["fp16"],
-                "no_post_process": config["no_post_process"],
                 "device": config["device"],
             }
         )
-        hf_config = deepcopy(model.hf_config) or HfConfig()
-        if version.parse(optimum_version.__version__) >= version.parse("1.14.0"):
-            # Optimum 1.14.0 needs to be run in legacy mode to support older versions of transformers
-            # TODO(trajep): deprecated legacy after fully test with the model using optimum merging
-            config["extra_args"]["legacy"] = True
-
-        export_optimum_model(
-            model.model_path or hf_config.model_name,
-            output_model_path,
-            **config["extra_args"],
-        )
+        hf_config = model.hf_config or HfConfig()
+        if hf_config.from_pretrained_args and "trust_remote_code" not in extra_args:
+            extra_args["trust_remote_code"] = hf_config.from_pretrained_args.trust_remote_code
 
-        hf_component_names = [name for name, _ in model.get_hf_components()]
-        if len(hf_component_names) == 0:
-            return ONNXModelHandler(resolve_onnx_path(output_model_path))
-        elif len(hf_component_names) == 1:
-            return ONNXModelHandler(Path(output_model_path) / hf_component_names[0])
-
-        model_components = []
-        model_component_names = []
-        for component_name in hf_component_names:
-            model_components.append(
-                ONNXModelHandler(str(Path(output_model_path) / component_name), model_attributes=model.model_attributes)
+        if version.parse(optimum_version.__version__) < version.parse("1.14.0"):
+            logger.warning(
+                "The behavior of Optimum onnx exporter changed in version 1.14.0 with the introduction of `legacy`"
+                " option. You are using an older version of optimum so it will use the legacy behavior and the output"
+                " model/s may not be the same as the latest version. Please upgrade to the latest version of optimum if"
+                " you do not want the legacy behavior!"
             )
-            model_component_names.append(component_name)
+            if "legacy" in extra_args:
+                logger.warning(
+                    "`legacy` option is set in the extra_args, but it is ignored because you are using optimum<1.14.0."
+                )
+                del extra_args["legacy"]
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            export_optimum_model(
+                model.model_path or hf_config.model_name,
+                temp_dir,
```

Comment:
>temp_dir [](http://example.com/codeflow?start=16&length=8)

do we need cache the decoder/decoder_past intermediate model for troubleshooting purpose?

Reply:
removed tempfolder as discussed offline.